src/frontend/hooks/useLocationProviderStatus.ts

```diff
@@ -21,6 +23,8 @@ export function useLocationProviderStatus() {
       getProviderStatusAsync()
         .then(status => {
           if (ignore) return;
+          if (!status.locationServicesEnabled)
+            queryClient.invalidateQueries({queryKey: ['lastLocation']});
```

Comment:
if location is no longer available, update the cache with the most recent last saved location